### PR TITLE
[mdatagen] add LeveledMeter method

### PR DESCRIPTION
**Pull request by codeboten**
This will use the LeveledMeterProvider in TelemetrySettings.
    
Split from https://github.com/open-telemetry/opentelemetry-collector/pull/10912, follows #10931

### CODE REVIEW

File: .chloggen/codeboten_leveled-meter-split-2.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+change_type:
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: component
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Add `TelemetrySettings.LeveledMeterProvider` func to replace MetricsLevel in the near future"
+
+# One or more tracking issues or pull requests related to the change
+issues: []
```
Comment (codeboten) [inline .chloggen/codeboten_leveled-meter-split-2.yaml:13]:
```suggestion
issues: [10933]
```

File: .chloggen/codeboten_leveled-meter-split-3.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: mdatagen
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Add `LeveledMeter` method to mdatagen"
+
+# One or more tracking issues or pull requests related to the change
+issues: []
```
Comment (codeboten) [inline .chloggen/codeboten_leveled-meter-split-3.yaml:13]:
```suggestion
issues: [10933]
```

**Review by mx-psi — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Comment by dmitryax** [comment]
This change makes the `LeveledMeterProvider` field in `TelemetrySettings` required. If not provided, users get panic:
```
ic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102e1320c]

goroutine 16 [running]:
testing.tRunner.func1.2({0x107ccdf80, 0x10cb4a620})
	/opt/homebrew/Cellar/go@1.22/1.22.7/libexec/src/testing/testing.go:1631 +0x1c4
testing.tRunner.func1()
	/opt/homebrew/Cellar/go@1.22/1.22.7/libexec/src/testing/testing.go:1634 +0x33c
panic({0x107ccdf80?, 0x10cb4a620?})
	/opt/homebrew/Cellar/go@1.22/1.22.7/libexec/src/runtime/panic.go:770 +0x124
go.opentelemetry.io/collector/receiver/receiverhelper/internal/metadata.LeveledMeter({0x14002da8080, {0x108d46498, 0x14002d9d2c0}, {0x108d464c0, 0x14002d9d2d0}, 0x0, 0x2, {0x0, 0x0}}, 0x186?)
	/Users/danoshin/Projects/opentelemetry-collector/receiver/receiverhelper/internal/metadata/generated_telemetry.go:21 +0x3c
go.opentelemetry.io/collector/receiver/receiverhelper/internal/metadata.NewTelemetryBuilder({0x14002da8080, {0x108d46498, 0x14002d9d2c0}, {0x108d464c0, 0x14002d9d2d0}, 0x0, 0x2, {0x0, 0x0}}, {0x0, ...})
	/Users/danoshin/Projects/opentelemetry-collector/receiver/receiverhelper/internal/metadata/generated_telemetry.go:51 +0x128
go.opentelemetry.io/collector/receiver/receiverhelper.newReceiver({{{{0x14002bb19b0, 0x9}}, {0x14002bb19ba, 0xd}}, {0x1055abc97, 0x4}, 0x0, {{{{...}}, {0x14002bb19ba, 0xd}}, ...}})
	/Users/danoshin/Projects/opentelemetry-collector/receiver/receiverhelper/obsreport.go:52 +0x54
go.opentelemetry.io/collector/receiver/receiverhelper.NewObsReport(...)
	/Users/danoshin/Projects/opentelemetry-collector/receiver/receiverhelper/obsreport.go:48
```

Is the new field expected to be required? If so, should it be marked as a breaking change?

**Comment by dmitryax** [comment]
@codeboten let me know if we need an issue from this? I can create one

**Comment by codeboten** [comment]
This is expected as the LeveledMeterProvider *should* be present. Adding a note to the changelog makes sense

**Pull request merged**